builder/dockerfile/internals.go

```diff
@@ -298,3 +340,39 @@ func hostConfigFromOptions(options *types.ImageBuildOptions) *container.HostConf
 		ExtraHosts: options.ExtraHosts,
 	}
 }
+
+func containsWildcards(name, platform string) bool {
+	for i := 0; i < len(name); i++ {
+		ch := name[i]
+		if ch == '\\' && platform != "windows" {
+			i++
+		} else if ch == '*' || ch == '?' || ch == '[' {
+			return true
+		}
+	}
+	return false
+}
+
+func validateCopySourcePath(imageSource *imageMount, origPath, platform string) error {
```

Comment:
Please move these two helpers to below the single function where they are called.